File: app/src/main/java/dev/chu/architecture/java/p80/sub_component/Machine.java
//package dev.chu.architecture.java.p80.sub_component;
//
//public class Machine {
//
//    private MachineComponent component;
//
//    public Machine(MachineComponent.Builder builder) {
//        component = builder.setMachineModule(new MachineModule())
//                .build();
//        component.inject(this);
//    }
//
//    public Coffee extract() {
//        return component.getCoffee();
//    }
//
//}

File: app/src/main/java/dev/chu/architecture/kotlin/p80/sub_component/Machine.kt
package dev.chu.architecture.kotlin.p80.sub_component

class Machine(builder: MachineComponent.Builder) {

    private var component: MachineComponent = builder.setMachineModule(MachineModule())
        .build()

    init {
        component.inject(this)
    }

    fun extract(): Coffee = component.getCoffee()
}
